src/main/scala/io/iohk/ethereum/ledger/Ledger.scala

```diff
@@ -303,17 +303,28 @@ class LedgerImpl(vm: VM, blockchain: BlockchainImpl, blockchainConfig: Blockchai
     worldStateProxy.saveAccount(senderAddress, account.increaseBalance(-calculateUpfrontGas(stx.tx)).increaseNonce)
   }
 
-  private[ledger] def prepareProgramContext(stx: SignedTransaction, blockHeader: BlockHeader, worldStateProxy: InMemoryWorldStateProxy, config: EvmConfig): PC =
-    stx.tx.receivingAddress match {
-      case None =>
-        val address = worldStateProxy.createAddress(creatorAddr = stx.senderAddress)
-        val world1 = worldStateProxy.transfer(stx.senderAddress, address, UInt256(stx.tx.value))
-        ProgramContext(stx, address,  Program(stx.tx.payload), blockHeader, world1, config)
+  private[ledger] def prepareProgramContext(stx: SignedTransaction, blockHeader: BlockHeader, worldStateProxy: InMemoryWorldStateProxy,
+                                            config: EvmConfig): PC = {
+    def prepareContext: PC = {
```

Comment:
What is the point of this closure?

Reply:
I like to have logic in separate function if logic in it is over 5LOC. After simplification (introduced in previous comment) it won't be necessary i think